go-controller/pkg/util/subnet_annotations.go

```diff
@@ -43,21 +44,62 @@ const (
 	ovnNodeLocalNatIP = "k8s.ovn.org/node-local-nat-ip"
 )
 
+func updateSubnetAnnotation(annotations map[string]string, annotationName, netName string, hostSubnets []*net.IPNet) error {
```

Comment:
this function is a bit complex to follow, appreciate if you can add a comment explaining what is doing

Reply:
@aojea Thanks for reviewing this. I've addressed your comments and repushed.  At this time, the changes is not fully ready for review though, will let you know once it is ready.